core/src/test/java/org/apache/sling/testing/mock/osgi/OsgiServiceUtilTest.java

```diff
@@ -100,7 +100,6 @@ public void testService3() {
 
         List<Map<String, Object>> reference3Configs = service3.getReference3Configs();
         assertEquals(1, reference3Configs.size());
-        assertEquals(200, reference3Configs.get(0).get(Constants.SERVICE_RANKING));
```

Comment:
why are you removing this test (and the testWithOsgiMetadata test case above)?

i think because the call to `MapMergeUtil.propertiesMergeWithOsgiMetadata` was moved to MockOsgi and out of MockBundleContext it is not longer applied to all code paths - e.g. when registering a service instance directly with the bundle context without instantiating or activating it via MockOsgi (e.g. a mockito mock object). that's why those test cases are failing.

Reply:
That was a deliberate change. `MockOsgi.newBundleContext().registerService(...)` must not consider metadata or ConfigAdmin data to be in line with regular OSGi Containers. This call is defined by the framework (http://docs.osgi.org/specification/osgi.core/7.0.0/framework.service.html#framework.service.registeringservices) and does not have any relation/dependency on any Compendium Chapters (like http://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.component.html,  http://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.cm.html or http://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.metatype.html).

What makes you think that component descriptions (http://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.component.html#d0e38089), Meta type resource (http://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.metatype.html#i1504250) nor configAdmin are relevant in this context?

Reply:
we currently do not have that clear distinction if a service in osgi mock is registered via DS with all it's metadata magic or not. it may be there is a test code out there that expects this metadata detection "just works" on whatever way you register the component in the mock context. this would be a breaking change then, although formally correct, leaving people who update to latest version of mocks wondering why it's not working any longer. so i'm wondering if it would do harm if we keep it as it was before - applying the DS magic everywhere, even if only BundleContext.registerService is called.
if we decide differently, we should create a separate ticket and PR for this change and document it properly, no one would expect such a change in a ticket which adds support for those two new properties.

Reply:
> so i'm wondering if it would do harm if we keep it as it was before - applying the DS magic everywhere, even if only BundleContext.registerService is called.

Yes, that may cause harm as then the mock testing may succeed because it considers OSGi metadata but real code does not work because it uses BundleContext.registerService(). Also the other way around is possible, e.g. test code fails, because it relies incorrectly on DS/metatype metadata while on real OSGi container the properties are not set.

Reply:
Backwards-compatibility is not a good reason in a mocking framework to not comply with the specs of the interface it tries to mock :-)

Reply:
> if we decide differently, we should create a separate ticket and PR for this change and document it properly, no one would expect such a change in a ticket which adds support for those two new properties.

Agree, I created https://issues.apache.org/jira/browse/SLING-10922 and will prepare a PR for that.

Reply:
thanks - then we can split it up and the new behavior more obvious in the changelog